cmd/config/internal/commands/util_test.go

```diff
@@ -0,0 +1,116 @@
+package commands
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+)
+
+func TestExecuteCmdOnPkgs(t *testing.T) {
+	var tests = []struct {
+		name        string
+		recurse     bool
+		pkgPath     string
+		executeCmd  func(string) error
+		needOpenAPI bool
+		errMsg      string
+	}{
+		{
+			name:        "need_Krmfile_error",
+			recurse:     true,
+			needOpenAPI: true,
+			pkgPath:     "subpkg1/subdir1",
+			executeCmd: func(string) error {
+				return nil
+			},
+			errMsg: `unable to find "Krmfile" in package "${baseDir}/subpkg1/subdir1"`,
+		},
+		{
+			name:        "Krmfile_not_needed_no_err",
+			recurse:     true,
+			needOpenAPI: false,
+			pkgPath:     "subpkg1/subdir1",
+			executeCmd: func(string) error {
+				return nil
+			},
+		},
+		{
+			name:        "executeCmd_returns_error",
+			recurse:     true,
+			needOpenAPI: true,
+			executeCmd: func(pkgPath string) error {
+				return errors.Errorf("this is a weird command which always returns an error")
+			},
+			errMsg: `this is a weird command which always returns an error`,
+		},
+	}
+
+	dir, err := ioutil.TempDir("", "")
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	defer os.RemoveAll(dir)
+	err = createTestDirStructure(dir)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			err := executeCmdOnPkgs(filepath.Join(dir, test.pkgPath), test.recurse, test.needOpenAPI, test.executeCmd)
+			if test.errMsg == "" {
+				if !assert.NoError(t, err) {
+					t.FailNow()
+				}
+			} else {
+				if !assert.Error(t, err) {
+					t.FailNow()
+				}
+				expectedError := strings.Replace(test.errMsg, "${baseDir}", dir, -1)
+				if !assert.Equal(t, expectedError, err.Error()) {
+					t.FailNow()
+				}
+			}
+		})
+	}
+}
+
+func createTestDirStructure(dir string) error {
+	/*
+		Adds the folders to the input dir with following structure
```

Comment:
Should we also try to test packages without any nested subpackages and several layers of nesting?

Reply:
Done.